### DISCUSSION TITLE
using EventSource server side

### DISCUSSION THREAD

**Discussion by dagda1** [denoland/deno#16264] [Q&A]
I wanted to use `EventSource` server side but looking at [this code](https://deno.land/x/eventsource@v0.0.2/mod.ts?source=#L72), it for some reason is reliant on the `window` object which is not the case in node.

I have a cli tool and when I create an `EventSource` object, I get

> SyntaxError: Cannot read properties of undefined (reading 'href')****

Is there anyway I can use the EventSource class on the server?

**Comment by ayame113** [comment]
Deno has a windows object.

I think this error is probably caused by window.location being undefined.
I think passing the `--location` flag at runtime will work.

```ts
import { EventSource } from "https://deno.land/x/eventsource@v0.0.2/mod.ts";
new EventSource("https://foobar.com/");
```

```shell
> deno run --location=http://foo.com --allow-net ./test.ts
```

Possibly related: https://github.com/MierenManz/EventSource/issues/30

-----

EDIT: Rereading the question, are you trying to use this with node? If so, sorry I'm not sure.

**Comment by dagda1** [comment]
@ayame113 no, this is not node, this is Deno and you are right, `--location` switch was the answer.

I had no idea there was a window object.  This is good to know
